iocore/net/quic/QUICLossDetector.cc

```diff
@@ -177,15 +179,14 @@ QUICLossDetector::update_ack_delay_exponent(uint8_t ack_delay_exponent)
 
 void
 QUICLossDetector::_on_packet_sent(QUICPacketNumber packet_number, bool ack_eliciting, bool in_flight, bool is_crypto_packet,
-                                  size_t sent_bytes, QUICPacketUPtr packet)
+                                  size_t sent_bytes, QUICPacketType type, std::vector<QUICFrameInfo> &frames)
 {
```

Comment:
Since we don't keep packet now, Can we make `frames` out of `QUICPacket`

Reply:
Sounds like a good idea

 I'm thinking of making these two changes:
1) Create a `PacketInfoUPtr` in `_state_common_send_packet` and pass(move) it to `on_packet_sent`
2) Pass `std::vector<QUICFrameInfo>` in a `PacketInfo` to `_packetize_frames` as a reference

So `frames` is a member of `PacketInfo` from the beginning to the end. This would remove the magical `std::move(frames)`.

Reply:
Sounds good.